install/kubernetes/helm/istio/templates/sidecar-injector-configmap.yaml

```diff
@@ -151,13 +151,14 @@ data:
           {{ "[[ else -]]" }}
           runAsUser: 1337
           {{ "[[- end ]]" }}
+        {{ "[[ if (isset .ObjectMeta.Annotations `sidecar.istio.io/proxyCPU`) -]]" }}
         resources:
-          {{ "[[ if (isset .ObjectMeta.Annotations `sidecar.istio.io/proxyCPU`) -]]" }}
           requests:
             cpu: {{ "\"[[ index .ObjectMeta.Annotations `sidecar.istio.io/proxyCPU` ]]\"" }}
             memory: {{ "\"[[ index .ObjectMeta.Annotations `sidecar.istio.io/proxyMemory` ]]\"" }}
         {{ "[[ else -]]" }}
 {{- if .Values.global.proxy.resources }}
+        resources:
```

Comment:
resources here is duplicated twice.  That might be ok, because the sidecar injector helm code uses nested gotpl (which is hard to read).

Reply:
This looks like a duplicate field. This probably needs an if/elseif, e.g.


if /* annotation set */ 
    use annotation overrides
elseif /* values.global.proxy.resources */
    use global resources
end

Reply:
@ayj HI, if don't make the /*resources*/ field duplicate, maybe could not resolve the problem above which I have mentioned, because the first outer /*IF*/ must align with the /*requests*/ field, it should indents two spaces. if the ELSE becomes empty, it will generate a wrong yaml format, cause the /*volumeAmount*/ filed indents two spaces.

Reply:
AFAICK, the gotemplate conditionals don't need to be aligned with the underlying YAML. You can adjust the indentation of the inserted yaml below be changed `indent 10` to whatever is required. Duplicating the resource will cause one of the values to be overwritten.

Reply:
@ayj HI, you may be mistaken, the gotemplate conditionals like /*{{ "[[ if (isset .ObjectMeta.Annotations `sidecar.istio.io/proxyCPU`) -]]" }}*/ indeed need to be aligned with the underlying YAML . you could try by indent it forwards or backwards, both does not work fine again. So far, I still could not find a way to resovle it beyond duplicating the resources field,  hopefully someone to give a great solution or replace this pr,  thanks